presto-client/src/main/java/com/facebook/presto/client/JsonResponse.java

```diff
@@ -145,6 +146,10 @@ public String toString()
             return new JsonResponse<>(response.code(), response.message(), response.headers(), body);
         }
         catch (IOException e) {
+            // OkHttp throws this after clearing the interrupt status
+            if ((e instanceof InterruptedIOException) && "thread interrupted".equals(e.getMessage())) {
```

Comment:
- is check on message necessary?
- can InterruptedIOException be wrapped in another IOException ?

Reply:
See the second part of `throwIfReached()` in `Timeout`. It throws `InterruptedIOException` for timeout as well. In theory it could be wrapped, but I didn't see that in the tests, and it seems unlikely given that the client `execute()` method directly throws `IOException`.

But these are good points, thanks for thinking of them. This whole thing is quite hacky.